MetadataExtractor/Formats/Jpeg/HuffmanTablesDirectory.cs

```diff
@@ -0,0 +1,323 @@
+            0xF1, 0xF2, 0xF3, 0xF4, 0xF5, 0xF6, 0xF7, 0xF8,
+            0xF9, 0xFA
+        };
+
+        internal static byte[] TypicalChrominanceAcLengths = {
+            0x00, 0x02, 0x01, 0x02, 0x04, 0x04, 0x03, 0x04,
+            0x07, 0x05, 0x04, 0x04, 0x00, 0x01, 0x02, 0x77
+        };
+
+        internal static byte[] TypicalChrominanceAcValues = {
+            0x00, 0x01, 0x02, 0x03, 0x11, 0x04, 0x05, 0x21,
+            0x31, 0x06, 0x12, 0x41, 0x51, 0x07, 0x61, 0x71,
+            0x13, 0x22, 0x32, 0x81, 0x08, 0x14, 0x42, 0x91,
+            0xA1, 0xB1, 0xC1, 0x09, 0x23, 0x33, 0x52, 0xF0,
+            0x15, 0x62, 0x72, 0xD1, 0x0A, 0x16, 0x24, 0x34,
+            0xE1, 0x25, 0xF1, 0x17, 0x18, 0x19, 0x1A, 0x26,
+            0x27, 0x28, 0x29, 0x2A, 0x35, 0x36, 0x37, 0x38,
+            0x39, 0x3A, 0x43, 0x44, 0x45, 0x46, 0x47, 0x48,
+            0x49, 0x4A, 0x53, 0x54, 0x55, 0x56, 0x57, 0x58,
+            0x59, 0x5A, 0x63, 0x64, 0x65, 0x66, 0x67, 0x68,
+            0x69, 0x6A, 0x73, 0x74, 0x75, 0x76, 0x77, 0x78,
+            0x79, 0x7A, 0x82, 0x83, 0x84, 0x85, 0x86, 0x87,
+            0x88, 0x89, 0x8A, 0x92, 0x93, 0x94, 0x95, 0x96,
+            0x97, 0x98, 0x99, 0x9A, 0xA2, 0xA3, 0xA4, 0xA5,
+            0xA6, 0xA7, 0xA8, 0xA9, 0xAA, 0xB2, 0xB3, 0xB4,
+            0xB5, 0xB6, 0xB7, 0xB8, 0xB9, 0xBA, 0xC2, 0xC3,
+            0xC4, 0xC5, 0xC6, 0xC7, 0xC8, 0xC9, 0xCA, 0xD2,
+            0xD3, 0xD4, 0xD5, 0xD6, 0xD7, 0xD8, 0xD9, 0xDA,
+            0xE2, 0xE3, 0xE4, 0xE5, 0xE6, 0xE7, 0xE8, 0xE9,
+            0xEA, 0xF2, 0xF3, 0xF4, 0xF5, 0xF6, 0xF7, 0xF8,
+            0xF9, 0xFA
+        };
+        
+        private static readonly Dictionary<int, string> _tagNameMap = new Dictionary<int, string>
+        {
+            { TagNumberOfTables, "Number of Tables" }
+        };
+
+        public HuffmanTablesDirectory()
+        {
+            SetDescriptor(new HuffmanTablesDescriptor(this));
+        }
+
+        public override string Name => "Huffman";
+
+        protected override bool TryGetTagName(int tagType, out string tagName)
+        {
+            return _tagNameMap.TryGetValue(tagType, out tagName);
+        }
+
+        /// <remarks>Use GetNumberOfTables for bounds-checking.</remarks>
+        /// <param name="tableNumber">The zero-based index of the table. This number is normally between 0 and 3.</param>
+        /// <returns>The HuffmanTable having the specified number.</returns>
+        [NotNull]
+        public HuffmanTable GetTable(int tableNumber)
+        {
+            return Tables[tableNumber];
+        }
+
+        /// <returns>The number of Huffman tables held by this HuffmanTablesDirectory instance.</returns>
+        public int GetNumberOfTables()
+        {
+            return this.GetInt32(TagNumberOfTables);
+        }
+
+        /// <returns>The List of HuffmanTables in this Directory.</returns>
+        [NotNull]
+        public List<HuffmanTable> Tables { get; set; } = new List<HuffmanTable>(4);
```

Comment:
I wouldn't make a mutable `List<>` public here. Is this needed, given you've added `GetTable` and `GetNumberOfTables`? If so, you could make the list a field and change the property to return `IReadOnlyList<HuffmanTable>`.

Reply:
The only reason this property exists (from what I can tell) is so JpegDhtReader can add new tables and update TagNumberOfTables to the current count from outside the directory. I think I'll try making Tables private, adding a new AddTable(HuffmanTable) method, and updating the count inside of that method instead. That makes a public List property unnecessary with the only tradeoff that the value under the tag might get updated more than once.

Reply:
On the plus side the tag value should always be correct.